specs/models/Step.schema.json

```diff
@@ -132,6 +132,56 @@
     },
     "name": {
       "type": "string"
+    },
+    "properties": {
+      "type": "object",
+      "$ref": "./MappingProperties.schema.json"
+    },
+    "expressionContext": {
+      "type":  "string",
+      "description": "An element in the source data from which to derive the values of this entity's properties"
+    },
+    "uriExpression": {
+      "type":  "string",
+      "description": "XPath expression for generating the uri of the entity instance document"
+    },
+    "relatedEntityMappings": {
+      "type":  "array",
+      "description": "additional collections provided by the user that get applied to the step output",
+      "items": {
+        "type": "object",
+        "properties": {
+          "relatedEntityMappingId": {
+            "type": "string",
+            "description": "Unique identifier of the related entity mapping"
+          },
+          "expressionContext": {
+            "type":  "string",
+            "description": "An element in the source data from which to derive the values of this related entity's properties"
+          },
+          "uriExpression": {
+            "type":  "string",
+            "description": "XPath expression for generating the uri of this related entity instance document"
+          },
+          "properties": {
+            "type": "object",
+            "$ref": "./MappingProperties.schema.json"
+          },
+          "collections": {
+            "type": "array",
+            "description": "additional collections provided by the user that get applied to this entity instance document"
+          },
+          "permissions": {
+            "type": "string",
+            "description": "Comma-delimited string of role,capability,role,capability,etc for the entity instance generated by this mapping"
+          },
+          "targetEntityType": {
+            "type": "string",
+            "description": "The identifier of the related Entity Type. (IRI, with title as fallback)"
+          }
+        },
+        "required": ["relatedEntityMappingId", "properties", "uriExpression"]
```

Comment:
I think targetEntityType has to be required as well . I don't think uriExpression has to be required though - the mapper can always use a default approach if no URI expression is provided.